### ISSUE TITLE
Cannot build on rust nightly

### ISSUE THREAD

**Issue by ysimonson** [iron/params#11]
Hello! I tried forking the project to fix an error I'm seeing in one of params' dependencies, but building the fork without any changes is causing a different set of errors:

``` bash
$ cargo build
   Compiling params v0.0.3 (file:///home/ysimonson/params)
src/lib.rs:170:9: 170:29 error: mismatched types:
 expected `&mime::Mime`,
    found `&mime::Mime`
(expected struct `mime::Mime`,
    found a different struct `mime::Mime`) [E0308]
src/lib.rs:170         &self.0.content_type
                       ^~~~~~~~~~~~~~~~~~~~
src/lib.rs:170:9: 170:29 help: run `rustc --explain E0308` to see a detailed explanation
src/lib.rs:170:9: 170:29 note: Perhaps two different versions of crate `mime` are being used?
src/lib.rs:170         &self.0.content_type
                       ^~~~~~~~~~~~~~~~~~~~
src/lib.rs:515:59: 515:71 error: mismatched types:
 expected `&hyper::header::Headers`,
    found `&hyper::header::Headers`
(expected struct `hyper::header::Headers`,
    found a different struct `hyper::header::Headers`) [E0308]
src/lib.rs:515     let boundary = match formdata::get_multipart_boundary(&req.headers) {
                                                                         ^~~~~~~~~~~~
src/lib.rs:515:59: 515:71 help: run `rustc --explain E0308` to see a detailed explanation
src/lib.rs:515:59: 515:71 note: Perhaps two different versions of crate `hyper` are being used?
src/lib.rs:515     let boundary = match formdata::get_multipart_boundary(&req.headers) {
                                                                         ^~~~~~~~~~~~
```

For reference:

``` bash
$ rustc --version
rustc 1.9.0-nightly (02310fd31 2016-03-19)
```

Tried digging in further, but I'm not enough of a Rustacean to figure out what's going on.

**Comment by rix0rrr** [comment]
I'm having the same, and I'm not even on nightly:

```
$ rustc --version
rustc 1.6.0
```

**Comment by SkylerLipthay** [comment]
Working on this issue. It seems like the upstream `hyper 0.8` release and subsequent `iron 0.3` release caused some problems.

**Comment by russoue** [comment]
I am having the same issue on rustc 1.7.

**Comment by SkylerLipthay** [comment]
Fixed these dependencies and released 0.1.0 with c204b06ba876a79bcce7645bf6cdfaaa56ced401. Thanks for your patience, everyone!

(In retrospect, I should have made a PR, but I was in a hurry!)

**Comment by ysimonson** [comment]
Thank you!
